### PR TITLE
Server error handling

### LINKED ISSUE #61: Server error handling
Add error handling to the server:

- [x] 200 for success

- [x] 204 for success but no results found

- [x] 400 for bad request, URL parameters, date, etc.

- [x] 403 for not found, provided URL is not valid

- [x] 500 for server catch all

### PR DESCRIPTION
**Pull request by tgeeTechtonic**
### Type of change:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling - no new features

### Why is this change required? What problem(s) does it solve?
Added initial error handling to server endpoints to handle incomplete requests and more feedback on successful but empty returns from the database

### Were there any challenges while implementing this feature? If so, how were they addressed?
Adding the middleware piece was new, but fairly straightforward

### Link issues
Closes #61

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -4,18 +4,20 @@ const path = require('path').join(__dirname, '../client/dist/');
...
 app.use('/api', require('./routes/shoutoutRoutes'));
+app.use('/api', require('./routes/slackRoutes'));
 app.use('/api', require('./routes/userRoutes'));
 app.use('/api', require('./routes/valuesRoutes'));
-app.use('/api', require('./routes/slackRoutes'));
+app.all('/api*', (req, res) => {
+  res
+    .status(400)
+    .json('The endpoint you are trying to use has not been implemented');
+});
```
Comment (broxburyTechtonic) [inline server/app.js:21]: This json get sent back for all 400 status responses?

File: server/middleware/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  else next();
+};
+
+checkForMissingParams = (req, requiredParams) => {
+  let missingParams = [];
+  for (param of requiredParams) {
+    if (!Object.keys(req).includes(param)) missingParams.push(param);
+  }
+  return missingParams;
+};
```
Comment (broxburyTechtonic) [inline server/middleware/index.js:23]: NIce

**Review by broxburyTechtonic — APPROVED** [review]
Looks like a great start for the error handling. I wonder if we're ready for Jonathan to start testing some of this BE

File: server/app.js (hunk 1 above)
Comment (tgeeTechtonic) [reply server/app.js:21]: This is getting sent back for anything that hasn't been implemented, like /api/elliot isn't valid and therefore would get this response

**Review by emackinnonTechtonic — APPROVED** [review]

**Pull request merged**